src/GafferOSL/OSLObject.cpp

```diff
@@ -218,9 +264,50 @@ IECore::ConstObjectPtr OSLObject::computeProcessedObject( const ScenePath &path,
 		// Ignore the output color closure as the debug closures are used to define what is 'exported' from the shader
 		if( it->first != "Ci" )
 		{
-			outputPrimitive->variables[it->first] = PrimitiveVariable( PrimitiveVariable::Vertex, it->second );
+			outputPrimitive->variables[it->first] = PrimitiveVariable( interpolation, it->second );
 		}
 	}
 
 	return outputPrimitive;
 }
+
+void OSLObject::hash( const ValuePlug *output, const Context *context, IECore::MurmurHash &h ) const
+{
+	SceneElementProcessor::hash( output, context, h );
+
+	if( output == resampledNamesPlug() )
+	{
+		inPlug()->objectPlug()->hash( h );
+	}
+}
+
+void OSLObject::compute( Gaffer::ValuePlug *output, const Gaffer::Context *context ) const
+{
+	if( output == resampledNamesPlug() )
+	{
+		ConstPrimitivePtr prim = runTimeCast<const IECore::Primitive>( inPlug()->objectPlug()->getValue() );
+
+		if (!prim)
+		{
+			static_cast<StringPlug *>( output )->setValue( "" );
```

Comment:
`output->setToDefault()` might be _slightly_ better here, because it can reuse the internal default value rather than needing to construct a new one.